### PR TITLE
feat: allow publishing reports on infra.ci

### PR DESCRIPTION
**Pull request by dduportal**
This PR is related to https://github.com/jenkins-infra/infra-reports/pull/37.

It allows the method `publishReport()` to be executed from infra.ci and it relies on the `az` CLI that should already be installed on the agent template.

### CODE REVIEW

File: vars/publishReports.groovy [hunk 1]
```diff
@@ -8,57 +8,67 @@
...
-                    }
-                    // Blob container can be removed once files are uploaded on the azure file storage
-                    sh "az storage blob upload --account-name=prodjenkinsreports --container=reports --timeout=${timeout} --file=${filename} --name=${filename} ${uploadFlags}"
+        // Install the az CLI if not already present. TODO: remove once CLI installed in universal template image
+        sh '''
+        if [ ! command -v az >/dev/null 2>&1];
+        then
+            apk add --no-cache gcc musl-dev python3-dev libffi-dev openssl-dev cargo make \
+            && apk add --no-cache py3-pip py3-pynacl py3-cryptography \
+            && python3 -m pip install --no-cache-dir azure-cli
```
Comment (timja) [inline vars/publishReports.groovy:24]: this is probably downloading around 400mb or so each build so I wouldn't keep this for long if you can avoid it

Comment (dduportal) [reply vars/publishReports.groovy:24]: yep, I've added until the `jenkins-infra/docker-helmfile` with the az CLI is available for the build. Thanks for the tip

Comment (dduportal) [reply vars/publishReports.groovy]: removed

**Comment by dduportal** [comment]
> My apologies in advance for what must be a particularly harsh merge conflict. Let me know if you want me to resolve it for you.

Many thanks for the notifications and the work! No worries we'll handle this: that would be a great practical exercise with git conflicts.

File: vars/publishReports.groovy [hunk 2]
```diff
@@ -8,52 +8,55 @@
...
+          sh 'az storage blob upload --account-name=prodjenkinsreports --container=reports --timeout=${TIMEOUT} --file=${FILENAME} --name=${FILENAME} ${UPLOADFLAGS} --overwrite'
 
-          sh "az storage file upload-batch \
-                      --account-name prodjenkinsreports \
-                      --destination reports \
-                      --source ${dirname ?: '.'} \
-                      --destination-path ${dirname ?: '/'} \
-                      --pattern ${ basename ?: '*' } \
-                      ${uploadFlags}"
+          // `az storage file upload` doesn't support file uploaded in a remote directory that doesn't exist but upload-batch yes. Unfortunatly the cli syntax is a bit different and require filename and directory name to be set differently.
```
Comment (timja) [inline vars/publishReports.groovy:58]:
```suggestion
          // `az storage file upload` doesn't support file uploaded in a remote directory that doesn't exist but upload-batch yes. Unfortunately the cli syntax is a bit different and requires filename and directory name to be set differently.
```

**Review by timja — DISMISSED** [review]

**Review by lemeurherve — APPROVED** [review]

**Comment by dduportal** [comment]
Post note: merging this PR caused the problem https://github.com/jenkins-infra/repository-permissions-updater/issues/2520.

**Comment by dduportal** [comment]
Post note: same problem on https://github.com/jenkins-infra/core-taglibs-report-generator/issues/11

**Pull request merged**